### PR TITLE
[Bugfix] Fix decomposing utf8

### PR DESCRIPTION
**Pull request by dritter**
This was originally discovered by @ZeGentzy in #1338 .

Here we now decompose UTF-8 chars that might be set by the user.

@ZeGentzy could you confirm this works for you?

### CODE REVIEW

**Comment by ZeGentzy** [comment]
It does not.

```
	P9K_MULTILINE_LAST_PROMPT_PREFIX_ICON="%K{004}%F{000} %D{\uF017 \uF073  %H:%M  %d.%m.%y} %f%k%F{004} %f"
	P9K_TIME_FORMAT="%D{\uF017 \uF073  %H:%M  %d.%m.%y}"
```

**Comment by dritter** [comment]
Ha! Could you try again? I forgot to expand user icons..

**Comment by romkatv** [comment]
These are very important fixes 👍

File: functions/utilities.zsh [hunk 1]
```diff
@@ -203,20 +203,59 @@ function p9k::defined() {
...
+#   Nothing.
+##
+function p9k::expand() {
+  (( $+parameters[$1] )) || return
+  local -a ts=("${=$(typeset -p $1)}")
+  shift ts
+  local x
+  for x in "${ts[@]}"; do
+    [[ $x == -* ]] || break
+    # Don't change readonly variables. Ideally, we shouldn't modify any variables at all,
```
Comment (romkatv) [inline functions/utilities.zsh]: Sorry, I've missed this when I skimmed the changes the first time. I don't recommend copying this hack. It has bad failure mode when it doesn't work and some unsavory side effects when it does.

**Pull request merged**